samplers/aws/xray/remote_sampler_config.go

```diff
@@ -0,0 +1,71 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xray
+
+import (
+	"time"
+)
+
+const (
+	defaultProxyEndpoint   = "127.0.0.1:2000"
+	defaultPollingInterval = 300
+)
+
+// SamplerOption is a function that sets config on the sampler
+type Option func(options *config)
+
+type config struct {
+	proxyEndpoint                string
+	samplingRulesPollingInterval time.Duration
+	logger                       Logger
```

Comment:
I see most usages using `log.Logger` though some `logr.Logger` too @Aneurysm9 is there a pattern for logs?

https://github.com/open-telemetry/opentelemetry-go/search?q=logger

Reply:
There's not standardized logging in go as of yet and we have tried to provide custom logging implementation here so that users can add their own logging logic.

Reply:
There is a facility for providing a `logr.Logger` that the SDK can use for internal logging, but there is as yet no mechanism for instrumentations or SDK extensions to access that instance.  It would probably be best here to also accept a `logr.Logger`, which provides a consistent interface and choice regarding underlying implementation.

Reply:
Okay here I will remove the custom logging and then only accepts `logr.Logger` because I think accepting both logger implementation does not makes sense since `log4.Logger` will include all utilities.